velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.cpp

```diff
@@ -58,6 +63,11 @@ void HdfsWriteFile::append(std::string_view data) {
   if (data.size() == 0) {
     return;
   }
+  // Check Whether the file is opened for write.
+  VELOX_CHECK_EQ(
+      hdfsFileIsOpenForWrite(hdfsFile_),
+      1,
+      "Write failure in HdfsWriteFile::append as the file is not opened");
```

Comment:
ditto. "Cannot append to HDFS file because file is not opened for write: {}", filePath_